csrc/assume.h

```diff
@@ -0,0 +1,15 @@
+#include <ir_all_nodes.h>
+
+// Return boolean values representing the conditional you want to assume
+
+namespace nvfuser::assume {
+
+// Assume that all tensor sizes appearing in `value` are positive. Return
+// nullptr if not applicable. For example:
+//   tensorsAreNotEmpty(ceilDiv(T0.size[0], 5) * T0.size[1])
+//     -> T0.size[0] > 0 && T0.size[1] > 0
```

Comment:
Unclear what this means. Tensor sizes are assumed to be positive, so what is returned? When is it considered not applicable?

Reply:
just updated the doc. It should return boolean predicates stating what you want to assume. For this example, the `value` is `ceilDiv(T0.size[0], 5) * T0.size[1]`, it depends on `T0.size[0]` and `T0.size[1]`, so the return value should be the boolean predicate `T0.size[0] > 0 && T0.size[1] > 0`.

Reply:
OK, so `tensorsAreNotEmpty` is the assumption you want to have, and the returned `Val` represents a specific condition that is necessary to make the assumption true within the given `Val`. 

Haven't looked at the implementation closely yet, but how do you know which sub `Val's in the given `Val` represent tensor sizes. In the example here, how do you know if `i1` may be defined as `i1 = T0.size[0]`?

Reply:
> OK, so `tensorsAreNotEmpty` is the assumption you want to have, and the returned `Val` represents a specific condition that is necessary to make the assumption true within the given `Val`

Yes, correct

I just do a DFS search from `value`, during the search, if I see a `NamedScalar` that is a tensor size, then I put it in the result, otherwise, I just ignore it.